### PR TITLE
extract cli install code block to fragment

### PR DESCRIPTION
**Pull request by drochetti**
**Notes:**

Currently the CLI install code snippet is replicated in 15+ files. A single change requires too many changes and it's hard to guarantee all files are updated, correct and new files will follow the correct content and format.

I extracted the install code block into a fragment and updated all the occurrences to the CLI install code.

**ARM / Apple M1**:

The Packaged CLI does not work on the new ARM-based architecture, the Apple M1. I added a call-out until that issue is fixed.

**Preview:**

NPM with `sudo` note:

Screen Shot 2021-01-27 at 10 46 05 AM

cURL/Mac ARM callout:

Screen Shot 2021-01-27 at 10 46 13 AM

Suggestions on the messaging are welcome.

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by attilah — APPROVED** [review]
This is a pretty neat optimization, LGTM!

One nit: can the `inline-fragment` be a self-closing element? - as it will never have child content.

```
<inline-fragment src="~/fragments/cli-install-block.md" />
```

**Review by jamesonwilliams — APPROVED** [review]
Nice!

File: docs/fragments/cli-install-block.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+```bash
+curl -sL https://aws-amplify.github.io/amplify-cli/install-win -o install.cmd && install.cmd
+```
+
+</amplify-block>
+
+</amplify-block-switcher>
+
+**Note:**: Because we're installing the Amplify CLI globally, you might need to run the command above with `sudo` depending on your system policies.
```
Comment (edwardfoyle) [inline docs/fragments/cli-install-block.md:29]: I would make this note only part of the npm install block. Don't want to encourage anyone to sudo curl a bash script!

Comment (drochetti) [reply docs/fragments/cli-install-block.md:29]: This note was there before, I'm not really sure why it was added but I assumed customers had issues before? Maybe @renebrandel or @swaminator can provide more context.

Comment (renebrandel) [reply docs/fragments/cli-install-block.md:29]: Correct. Only have this for the NPM block

**Comment by drochetti** [comment]
> This is a pretty neat optimization, LGTM!
> 
> One nit: can the `inline-fragment` be a self-closing element? - as it will never have child content.
> 
> ```
> <inline-fragment src="~/fragments/cli-install-block.md" />
> ```

hey @attilah IIRC there's a limitation in our engine that parses those tags when it comes to self-closing tags. I can't find instances of self-closing tags anywhere.

File: docs/fragments/cli-install-block.md (hunk 1 above)
Comment (drochetti) [reply docs/fragments/cli-install-block.md]: @renebrandel I addressed this in the most recent commit. The `sudo` note is part of the NPM block and the ARM note part of the cURL (Mac and Linux) block

**Review by renebrandel — APPROVED** [review]

**Comment by dabit3** [comment]
This is great improvement!

**Review by dabit3 — APPROVED** [review]

**Comment by jamesonwilliams** [comment]
Not for this PR, but on a somewhat related note:

Our upgrade instructions aren't married back to the new `cURL` approach. They only talk about updating via `npm`.

https://docs.amplify.aws/cli/usage/upgrade

**Pull request merged**